pandas-stubs/core/indexes/base.pyi

```diff
@@ -275,6 +274,16 @@ class Index(IndexOpsMixin[S1], ElementOpsMixin[S1]):
         name: Hashable = ...,
         tupleize_cols: bool = ...,
     ) -> IntervalIndex[Interval[Any]]: ...
+    @overload
+    def __new__(
```

Comment:
This was added too low, as you can see the tests are failing meaning that your changes have not changed how the type checker identifies the types of an index of a DatetimeIndex.
What you will likely need is to drop this and add `DatetimeIndex` at line 190 along the sequence and IndexOpsMixIn of datetime.